native/explorer/src/series.rs

```diff
@@ -1345,23 +1345,38 @@ pub fn s_trim(s1: ExSeries, pattern: Option<&str>) -> Result<ExSeries, ExplorerE
     };
 
     // replace only replaces the leftmost match, so we need to call it twice.
-    let s1 = ExSeries::new(s1.utf8()?.replace(pattern.as_str(), "")?.into());
-
     Ok(ExSeries::new(
-        s1.utf8()?.replace(pattern.as_str(), "")?.into(),
+        s1.utf8()?
+            .replace(pattern.as_str(), "")?
+            .replace(pattern.as_str(), "")?
+            .into(),
     ))
 }
 
 #[rustler::nif(schedule = "DirtyCpu")]
-pub fn s_trim_leading(s1: ExSeries) -> Result<ExSeries, ExplorerError> {
+pub fn s_trim_leading(s1: ExSeries, pattern: Option<&str>) -> Result<ExSeries, ExplorerError> {
     // There are no eager strip functions.
-    Ok(ExSeries::new(s1.utf8()?.replace(r#"^[ \s]+"#, "")?.into()))
+    let pattern = match pattern {
+        None => String::from(r#"^[ \s]+"#),
+        Some(string) => format!(r#"^[{}]+"#, &string),
```

Comment:
should this be a group like `^(str)+` rather than a list of chars `^[str]+`

Reply:
@philss what do you think here? currently passing `trim_leading("ab")`

remove all leading `b` and/or all leading `a`

so string `aaaabbbbhello` would become `hello`

doing a non capturing group like `^(?:ab)+` would remove nothing

Reply:
or we could add a guard only allowing strings of length 1 to be passed?

Reply:
@DeemoONeill oh, good point! I think people would expect it to remove "ab" as a whole. So I would change it to a group.
WDYT, @josevalim ?

Reply:
> should this be a group like ^(str)+ rather than a list of chars ^[str]+

Correct. That's how Elixir's trim_trailing/trim_leading behaves. We can also compare with how rstrip/lstrip behaves by adding tests with two letters. :)

Reply:
okie doke! that's tomorrows job :)

Reply:
well, that was unexpected. 

i wrote a two letter test with rstrip/lstrip and  `"abab"` will return `"hello"` on `"abhelloba"`.

I think matching the expr_function will probably cause the least confusion here

Reply:
`"nx_"` as the pattern will strip `"_nx"` too

Reply:
So it seems it behaves different from Elixir and the given string is a set of characters to remove. So I think we should move away from the `trim` name in Explorer and move it all to `strip`, `rstrip`, and `lstrip`. What do you think?

Reply:
That's probably a safer way to handle it, i've added in the docstring that it removes the set of characters provided. I'll do a rename